### PR TITLE
Guac integration

### PR DESCRIPTION
**Pull request by mpapierski**
Rebased, and finally managed to get dependencies right.

Its not meant to be merged as it would break MIPS (see #236), but might be valuable to get another round of reviews that were present in old integration attempt in #151.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ Status: Needs fraud detection implementation
 ### bounty_hunter
  A separate daemon from Rita designed to be run by channel bounty hunters on the internet. In a production Alteha network mesh devices would periodically upload their channel states to a bounty hunter. The bounty hunter will then watch the blockchain state and publish these channel states if an attempt at fraud was made. Claiming a small bounty and preventing channel fraud even when a device is knocked offline. 
  
- Status: Needs Parity integration and real channel states
+ Status: Needs Parity integration and real channel states (removed for now, adding back to guac_rs)
```
[the remark this answers is not in the document]
Comment (mpapierski) [reply README.md]: Done

File: althea_kernel_interface/src/lib.rs [hunk 2]
```diff
@@ -118,6 +132,38 @@ impl CommandRunner for TestCommandRunner {
 
         (&mut *self.run_command.lock().unwrap())(program.to_string(), args_owned)
     }
+    fn test_commands(&self, _test_name: &str, test_commands: &[(&str, &str)]) {
+        let mut commands_owned = Vec::new();
+        for (command, result) in test_commands {
+            commands_owned.push((command.to_string(), result.to_string()))
```
[the remark this answers is not in the document]
Comment (mpapierski) [reply althea_kernel_interface/src/lib.rs]: Interestingly it was removed and changed to set_mock for some reason. Coincidentally this seems to be a side effect of a merge.

File: rita/src/rita_common/dashboard/mod.rs [hunk 3]
```diff
@@ -42,17 +48,23 @@ impl Handler<GetOwnInfo> for Dashboard {
...
+            .send(GetOwnBalance {})
+            .from_err()
+            .and_then(|own_balance| match own_balance {
+                Ok(balance) => Ok(OwnInfo {
+                    balance: balance.as_u64() as i64,
+                    version: env!("CARGO_PKG_VERSION").to_string(),
                 }),
-        )
+                Err(e) => Err(e),
+            });
```
[the remark this answers is not in the document]
Comment (mpapierski) [reply rita/src/rita_common/dashboard/mod.rs]: Removed feature flag completely.

File: scripts/openwrt_build_mipsel.sh [hunk 4]
```diff
@@ -21,4 +21,4 @@ export OPENSSL_STATIC=1
 
 rustup target add mipsel-unknown-linux-musl
 
-cargo build --target mipsel-unknown-linux-musl ${PROFILE} ${FEATURES} -p rita --bin rita
+cargo build --target mipsel-unknown-linux-musl ${PROFILE} ${FEATURES} -p rita --bin rita --features "system_alloc"
```
[the remark this answers is not in the document]
Comment (mpapierski) [reply scripts/openwrt_build_mipsel.sh]: Removed. Not sure why it was here. Probably unintended side effect after rebase.

File: rita/Cargo.toml [hunk 5]
```diff
@@ -26,6 +26,7 @@ clu = { path = "../clu" }
 exit_db = { path = "../exit_db" }
 num256 = { path = "../num256" }
 settings = { path = "../settings" }
+guac_actix = { git = "https://github.com/mpapierski/guac_rs", branch = "6-cargo", optional = true}
```
[the remark this answers is not in the document]
Comment (mpapierski) [reply rita/Cargo.toml]: Done. Switched to althea-mesh/guac_rs with master branch. (After merging I'd consider going on crates.io with guac in similiar way as serde-rlp/clarity)

File: exit_db/migrations/2018-07-29-124240_add_eth_addr/up.sql [hunk 6]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE clients
```
[the remark this answers is not in the document]
Comment (mpapierski) [reply exit_db/migrations/2018-07-29-124240_add_eth_addr/up.sql]: Basically yes. I'm not sure if there is easy way to deal with this besides waiting for an update wth new ETH address from the client and update it.

File: integration-tests/rita.py [hunk 7]
```diff
@@ -59,7 +55,7 @@
         "exit_a": {
             "id": {
                 "mesh_ip": "fd00::5",
-                "eth_address": "0x0101010101010101010101010101010101010101",
+                "eth_address": "fd00::5",
```
[the remark this answers is not in the document]
Comment (mpapierski) [reply integration-tests/rita.py]: Seems like unintended side effect. I'll revert this chunk.

File: rita/src/rita_common/rita_loop/mod.rs [hunk 8]
```diff
@@ -92,8 +95,11 @@ impl Handler<Tick> for RitaLoop {
         } else {
             self.was_gateway = false
         }
-
         let start = Instant::now();
+
+        DebtKeeper::from_registry().do_send(DebtTick);
+        DebtKeeper::from_registry().do_send(SendUpdate {});
+
         ctx.spawn(
```
[the remark this answers is not in the document]
Comment (mpapierski) [reply rita/src/rita_common/rita_loop/mod.rs]: I'm not sure whats the difference between ctx.spawn, and Arbiter::spawn. Perhaps @kingoflolz can help figure it out. In this case Arbiter::spawn is not a straightforward replacement. I couldn't figure it out and not sure why:

```
106 |         Arbiter::spawn(
    |         ^^^^^^^^^^^^^^ the trait `futures::Future` is not implemented for `actix::fut::Then<actix::fut::Then<actix::fut::FutureWrap<actix::prelude::Request<rita_common::tunnel_manager::TunnelManager, rita_common::tunnel_manager::GetNeighbors>, rita_common::rita_loop::RitaLoop>, actix::fut::Then<actix::fut::FutureWrap<actix::prelude::Request<rita_common::traffic_watcher::TrafficWatcher, rita_common::traffic_watcher::Watch>, rita_common::rita_loop::RitaLoop>, actix::fut::Then<actix::fut::FutureWrap<actix::prelude::Request<guac_actix::PaymentController, guac_actix::Tick>, rita_common::rita_loop::RitaLoop>, actix::fut::FutureResult<(), _, rita_common::rita_loop::RitaLoop>, [closure@rita/src/rita_common/rita_loop/mod.rs:135:39: 138:34]>, [closure@rita/src/rita_common/rita_loop/mod.rs:125:31: 139:26 neigh:_]>, [closure@rita/src/rita_common/rita_loop/mod.rs:110:23: 140:18 start:_]>, _, _>`
```

File: exit_db/migrations/2018-07-29-124240_add_eth_addr/up.sql (hunk 6 above)
[the remark this answers is not in the document]
Comment (jkilpatr) [reply exit_db/migrations/2018-07-29-124240_add_eth_addr/up.sql]: clients continue to query the exit for state changes, maybe we can add key exchange in at some point. or do you think the payment code should have messages like 'identity request' etc

Actually we already send the new info with the hello messages we constantly send the exit as a liveness indicator.

**Pull request closed without merging**